### DISCUSSION TITLE
[Feat]: Support pyspark.sql.DataFrame

### DISCUSSION THREAD

**Discussion by Asm-Def** [Kanaries/pygwalker#23] [Ideas]
What technical route do you think is most appropriate?

**Comment by rwitzel** [comment]
[Pandas API on Spark](https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/index.html) and more lazy execution in the library.

**Reply by Asm-Def** [reply]
The actual problem might be the huge data size of common pyspark DataFrames in practice, which leads to the requirement for the embedded web application to interact with the IPython kernel and convert all the JS logic into calls to pandas APIs.
